src/server/node_manager.c

```diff
@@ -3097,6 +3112,10 @@ mcast_moms(struct work_task *ptask)
 			mtfd_replyhello_noinv = -1;
 			break;
 
+		case PS_CONNECT:
+			replyhello_psvr();
```

Comment:
All ur acks for send_resc_usages result in decrementing the pending acks! This is fine in general. Are there cases where this count could go berserk, and when it does, what operation will reset it to a clean state? 
If a peer server was sent a resc_update() but crashed and did not respond, when it comes back up, will the pending_acks for that server be cleared and started from 0?

Reply:
Counts are reset when the server receive a PS_CONNECT from one of the peer server. Then all the resource updates will be sent.

IS_CONNECT will be received in the following cases:
1. When a new server joins the cluster .
2. When an existing server gets restarted
3. When network fails and then server rejoins the cluster after the network recovery.

Application layer relies on the network layer in detecting network failure.

Reply:
ok sounds fine for now

Reply:
one more question: If just the network goes down and comes back up, (the peer server did not failed and restart), then will the peer serve send PS_CONNECT to fix the counts?

Reply:
If the application layer came to know about the network failure, it will close the peer server connection. The server will periodically attempt to re-connect and the initial exchange will get triggered upon reconnecting.